examples/suit_update: add ble documentation in README (#12391)
RIOT-OS/RIOT: examples/suit_update/README.md

Table of contents:

- [Prerequisites][prerequisites]

>   - [Ble Prerequisites][prerequisites-ble]

Reviewer: ```suggestion
  - [Ble][prerequisites-ble]
```

  using `make term`, but that requires an already set up tap interface.
  See [update] for more information.

> ### Ble Prerequisites

Reviewer: ```suggestion
### Ble
```

> [prerequisites-ble]: #BlePrerequisites

Reviewer: ```suggestion
[prerequisites-ble]: #Ble
```

        Device F0:36:27:6B:F1:8F Decathlon Dual HR
        Device 69:B3:82:0B:73:C9 69-B3-82-0B-73-C9
        Device 43:1A:39:CD:39:B9 43-1A-39-CD-39-B9
        Device E4:DD:E0:8F:73:65 RIOT-autoconn
    ...

- Once you have the address, you simply connect Linux to RIOT using the following
command:

    # Put your device address here...
    # Note: the 2 after the address denotes a BLE public random address, default
    #       used by `nimble_netif`
    echo "connect UU:VV:WW:XX:YY:ZZ 2" > /sys/kernel/debug/bluetooth/6lowpan_control

- Verify that the ble interface has been correctly created:

    $ ifconfig bt0

    ...
    bt0: flags=4161<UP,RUNNING,MULTICAST>  mtu 1280
            inet6 fe80::19:86ff:fe00:16ca  prefixlen 64  scopeid 0x20<link>
            unspec 00-19-86-00-16-CA-00-1E-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
            RX packets 330  bytes 22891 (22.8 KB)
            RX errors 0  dropped 0  overruns 0  frame 0
            TX packets 354  bytes 30618 (30.6 KB)
            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
    ...

- You should now be able to ping the device

    $ ping6 fe80::e4dd:e0ff:fe8f:7365%bt0

> If the optional configuration to set-up a routable prefix is not performed then

Reviewer: This paragraph needs the be rewritten.

    SUIT_COAP_SERVER=[fe80::19:86ff:fe00:16ca]
    SUIT_CLIENT=[fe80::e4dd:e0ff:fe8f:7365%bt0]

- [optional]: follow the guide for distributing a routable Prefix in
  [README.ipv6-over-ble.md](../../pkg/nimble/README.ipv6-over-ble.md).

If this was performed correctly then the `bt0` interface should now have a global
address:

    bt0: flags=4161<UP,RUNNING,MULTICAST>  mtu 1280
            inet6 2001:db8::19:86ff:fe00:16ca  prefixlen 64  scopeid 0x0<global>
            inet6 fe80::19:86ff:fe00:16ca  prefixlen 64  scopeid 0x20<link>
            inet6 2001:db8::b004:c58:891f:aa09  prefixlen 64  scopeid 0x0<global>
            unspec 00-19-86-00-16-CA-00-14-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
            RX packets 3  bytes 120 (120.0 B)
            RX errors 0  dropped 0  overruns 0  frame 0
            TX packets 34  bytes 3585 (3.5 KB)
            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

In this case the address to use for `SUIT_COAP_SERVER` can be either the EUI64
generated global address `[2001:db8::19:86ff:fe00:16ca]` or the random global address
`[2001:db8::b004:c58:891f:aa09]`.

> I for some reason this didn't work the you can manually set up an address for

Reviewer: ```suggestion
If for some reason this didn't work the you can manually set up an address for
```

the subnet:

    $ sudo ip address add 2001:db8::1/64 dev bt0

In this case the address used for `SUIT_COAP_SERVER` should be [`2001:db8::1`].

In either case the address used for `SUIT_CLIENT` should be the suffix of the link
local address for that device (`e4dd:e0ff:fe8f:7365` in our examples) and the
distributed prefix, i.e.: `SUIT_CLIENT=[2001:db8::e4dd:e0ff:fe8f:7365]`

### Start aiocoap-fileserver
[Start-aiocoap-fileserver]: #start-aiocoap-fileserver

> `aiocoap-fileserver` is used for hosting firmware's available for updates.

Reviewer: A bit unrelated.
```suggestion
`aiocoap-fileserver` is used for hosting the firmwares available for updates.
```

                  inet6 group: ff02::2
                  inet6 group: ff02::1
                  inet6 group: ff02::1:ff8f:7365
    ...

**NOTE 2:** Currently, Linux does not support 6LoWPAN neighbor discovery (which
RIOT uses per default with BLE), so RIOT needs to be compiled to use stateless
address auto configuration (SLAAC) -> `CFLAGS=-DGNRC_IPV6_NIB_CONF_SLAAC=1`.

- Use `bluetoothctl` on Linux to scan for the device. Once `bluetoothctl` has
  started, issue `scan on` to start scanning. The default name for the RIOT
  device is set to `RIOT-autoconn`, so you should see it pop up. You can also
  use `devices` to list scanned devices.

      ...
      $ bluetoothctl
          Agent registered
          [bluetooth]# scan on
          Discovery started
          [CHG] Controller F4:5C:89:9F:AC:7A Discovering: yes
          [CHG] Device E4:DD:E0:8F:73:65 RSSI: -49
          [CHG] Device 43:1A:39:CD:39:B9 RSSI: -94
      ...
      ...
          [bluetooth]# devices
          Device F0:36:27:6B:F1:8F Decathlon Dual HR
          Device 69:B3:82:0B:73:C9 69-B3-82-0B-73-C9
          Device 43:1A:39:CD:39:B9 43-1A-39-CD-39-B9
          Device E4:DD:E0:8F:73:65 RIOT-autoconn
      ...

> - Once you have the address, you simply connect Linux to RIOT using the following

Reviewer: ```suggestion
- Once you have the address, simply connect Linux to RIOT using the following
```

      ...

- You should now be able to ping the device

      $ ping6 fe80::e4dd:e0ff:fe8f:7365%bt0

- **optional**: follow the guide for distributing a routable Prefix in
  [README.ipv6-over-ble.md](../../pkg/nimble/README.ipv6-over-ble.md).

If this was performed correctly then the `bt0` interface should now have a global
address:

    bt0: flags=4161<UP,RUNNING,MULTICAST>  mtu 1280
            inet6 2001:db8::19:86ff:fe00:16ca  prefixlen 64  scopeid 0x0<global>
            inet6 fe80::19:86ff:fe00:16ca  prefixlen 64  scopeid 0x20<link>
            inet6 2001:db8::b004:c58:891f:aa09  prefixlen 64  scopeid 0x0<global>
            unspec 00-19-86-00-16-CA-00-14-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
            RX packets 3  bytes 120 (120.0 B)
            RX errors 0  dropped 0  overruns 0  frame 0
            TX packets 34  bytes 3585 (3.5 KB)
            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

In this case the address to use for `SUIT_COAP_SERVER` can be either the EUI64
generated global address `[2001:db8::19:86ff:fe00:16ca]` or the random global address
`[2001:db8::b004:c58:891f:aa09]`.

> If for some reason this didn't work the you can manually set up an address for

Reviewer: ```suggestion
If for some reason this didn't work, you can manually set up an address for
```

the subnet:

    $ sudo ip address add 2001:db8::1/64 dev bt0

In this case the address used for `SUIT_COAP_SERVER` should be [`2001:db8::1`].

Route traffic going towards your subnet through bt0:

    $ sudo route -A inet6 add 2001:db8::/64 dev bt0

In either case the address used for `SUIT_CLIENT` should be the suffix of the link
local address for that device (`e4dd:e0ff:fe8f:7365` in our examples) and the
distributed prefix, i.e.: `SUIT_CLIENT=[2001:db8::e4dd:e0ff:fe8f:7365]`

> If this optional step is skipped then in then `SUIT_COAP_SERVER` will be

Reviewer: ```suggestion
If this optional step is skipped then `SUIT_COAP_SERVER` will be
```

the link local address of the `bt0` interface and `SUIT_CLIENT` will be

> link local address of the device, with the interface specified. e.g:

Reviewer: ```suggestion
the link local address of the device, with the interface specified. e.g:
```
